pkg/controller/proxyconfig/validation.go

```diff
@@ -52,5 +57,82 @@ func (r *ReconcileProxyConfig) ValidateProxyConfig(proxyConfig *configv1.ProxySp
 		}
 	}
 
+	if isSpecReadinessEndpoints(proxyConfig) {
+		for _, endpoint := range proxyConfig.ReadinessEndpoints {
+			scheme, err := validation.URI(endpoint)
+			if err != nil {
+				return fmt.Errorf("invalid uri for endpoint %s: %v", endpoint, err)
+			}
+			switch {
+			case scheme == proxyHTTPScheme:
+				// TODO: Add case for proxyHTTPSScheme when CA support is merged.
+				if err := validateHTTPReadinessEndpoint(proxyConfig.HTTPProxy, endpoint); err != nil {
+					return fmt.Errorf("readinessEndpoint probe failed for endpoint %s", endpoint)
+				}
+			default:
+				// TODO: Update error to include proxyHTTPSScheme when CA support is merged.
+				return fmt.Errorf("readiness endpoints requires a %q uri sheme", proxyHTTPScheme)
+			}
+		}
+	}
+
 	return nil
 }
+
+// validateHTTPReadinessEndpoint validates an http readinessEndpoint endpoint.
+func validateHTTPReadinessEndpoint(httpProxy, endpoint string) error {
+	if err := validateHTTPReadinessEndpointWithRetries(httpProxy, endpoint, proxyProbeMaxRetries); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// validateHTTPReadinessEndpointWithRetries tries to validate an http
+// endpoint in a finite loop and returns the last result if it never succeeds.
+func validateHTTPReadinessEndpointWithRetries(httpProxy, endpoint string, retries int) error {
+	var err error
+	for i := 0; i < retries; i++ {
+		err = runHTTPReadinessProbe(httpProxy, endpoint)
```

Comment:
might make sense to include a sleep between retries, for a second.

Reply:
@bparees PR updated to include wait time between probes.